### ISSUE TITLE
`int` and `float` get saved as `np.int32` and `np.float64`, `list` is saved as `np.ndarray`

### ISSUE THREAD

**Issue by pheuer** [varchasgopalaswamy/prepper#2]
Currently python datatypes get converted to numpy datatypes during saving, so _technically_ you're not getting back the same thing you exported. 

E.g.  `int(1)` will be re-imported as `numpy.int32`

Similarly, lists are re-imported as `np.ndarray`

**Comment by varchasgopalaswamy** [comment]
Yeah, I'm not sure there is a good way around this. The H5py driver is going to give you back numpy arrays for datasets. Will it matter? I suppose there may be edge cases where users explicitly expect a list and use list-specific methods, but perhaps that should be guarded by a explicit `list(obj)` call first?

**Comment by pheuer** [comment]
I was solving this by saving the literal datatype (e.g. type(var)=='numpy.float32' or type(var)=='int') as an attribute, and then checking that when reading in to cast appropriately. 

But, I'm not sure it's a problem as long as we document it somewhere. If these files are `language agnostic` then an int32 is an int32, regardless of the specific internal python type?

**Comment by varchasgopalaswamy** [comment]
> But, I'm not sure it's a problem as long as we document it somewhere. 

I agree. Once there's a documentation of this behavior somewhere I'll close.

**Comment by varchasgopalaswamy** [comment]
I'll also point out that python should broadly follow duck typing. If an application cares deeply that an object is indeed a `list`, and not just something that can be iterated over, then it should probably force the conversion. Or at least, that's going to be my out from having a ton of handlers for this kind of problem.
